python/lsst/ip/diffim/zogy.py

```diff
@@ -330,748 +127,715 @@ def _computeVarianceMean(self, exposure):
+        # diffSubExposure.maskedImage = calibImg
+        diffSubExposure.maskedImage /= ftDiff.Fd
+
+        # Now the subExposure calibration is 1. everywhere
+        calibOne = afwImage.PhotoCalib(1.)
+        diffSubExposure.setPhotoCalib(calibOne)
+
+        # Set the PSF of this subExposure
+        psfImg = self.subExpPsf1.Factory(self.subExpPsf1.getDimensions())
+        psfImg.array = Pd
+        psfNew = measAlg.KernelPsf(afwMath.FixedKernel(psfImg))
+        diffSubExposure.setPsf(psfNew)
+
+        if ftDiff.S is not None:
+            S = self.inverseFftAndCropImage(
+                ftDiff.S, self.imgShape, np.logical_or(self.fftFullIm1.filtInf, self.fftFullIm2.filtInf),
+                np.logical_or(self.fftFullIm1.filtNaN, self.fftFullIm2.filtNaN),
+                dtype=self.subExposure1.image.dtype)
+            varPlaneS = self.inverseFftAndCropImage(
+                ftDiff.varPlaneS, self.imgShape,
+                np.logical_or(self.fftVarPl1.filtInf, self.fftVarPl2.filtInf),
+                np.logical_or(self.fftVarPl1.filtNaN, self.fftVarPl2.filtNaN),
+                dtype=self.subExposure1.variance.dtype)
+
+            S = S[arrIndex.getY():arrIndex.getY() + bbox.getHeight(),
+                  arrIndex.getX():arrIndex.getX() + bbox.getWidth()]
+            varPlaneS = varPlaneS[arrIndex.getY():arrIndex.getY() + bbox.getHeight(),
+                                  arrIndex.getX():arrIndex.getX() + bbox.getWidth()]
+            # PSF of S
+            Ps = self.inverseFftAndCropImage(ftDiff.Ps, self.psfShape1, dtype=self.subExpPsf1.dtype)
+            sumPs = np.sum(Ps)
+            self.log.info(f"Ps sum before normalization: {sumPs:.3f}")
+            Ps /= sumPs
+
+            # Ensure that no 0/0 occur in S/var(S).
+            tiny = np.finfo(varPlaneS.dtype).tiny * 10
+            fltZero = np.nonzero(varPlaneS < tiny)
+            varPlaneS[fltZero] = tiny
+            S[fltZero] = 0
+
+            # TODO DM-23855 : Scorr corrections may be done here
+
+            scoreSubExposure = self.subExposure1.clone()
+            scoreSubExposure.image.array = S
+            scoreSubExposure.variance.array = varPlaneS
+            scoreSubExposure.mask = diffSubExposure.mask
+            scoreSubExposure.setPhotoCalib(None)
+
+            # Set the PSF of this subExposure
+            psfSImg = self.subExpPsf1.Factory(self.subExpPsf1.getDimensions())
+            psfSImg.array = Ps
+            psfSNew = measAlg.KernelPsf(afwMath.FixedKernel(psfSImg))
+            scoreSubExposure.setPsf(psfSNew)
         else:
-            res = task.computeScorr(**kwargs)
-            D = res.S
+            scoreSubExposure = None
 
-        outExp = D.Factory(D, subExposure.getBBox())
-        out = pipeBase.Struct(subExposure=outExp)
-        return out
+        return pipeBase.Struct(diffSubExp=diffSubExposure, scoreSubExp=scoreSubExposure)
 
+    def run(self, exposure1, exposure2, calculateS=True):
```

Comment:
Please rename `calculateS` to `calculateScorr` (or `calculateScore` if that's what the variable is called)